javascript/ql/src/semmle/javascript/frameworks/D3.qll

```diff
@@ -0,0 +1,82 @@
+/** Provides classes and predicates modelling aspects of the `d3` library. */
+private import javascript
+private import semmle.javascript.security.dataflow.Xss
+
+/** Provides classes and predicates modelling aspects of the `d3` library. */
+module D3 {
+  /** Gets an API node referring to the `d3` module. */
+  API::Node d3() {
+    result = API::moduleImport("d3")
+    or
+    result = API::moduleImport("d3-node").getInstance().getMember("d3")
+  }
```

Comment:
I had expected that `d3` would also be used as a global variable.

Reply:
Good point, we should totally recognize that.